astropy/io/ascii/ui.py

```diff
@@ -627,6 +627,30 @@ def _guess(table, read_kwargs, format, fast_reader):
         cparser.CParserError,
     )
 
+    # Check what line endings are being used in file
+    line_ending = "\r\n" if "\r" in table else "\n"
```

Comment:
In the more common case of `\n` line endings, this requires scanning the entire text. While that is relatively fast, is this handling of the line ending really necessary? 

If the later code just does `table.index("\n", pos + 1)`, then it seems like that will appropriately count lines (in either case) and then on Windows just end up with a dangling `\r`. I *think* that is OK and it seems to work, but I'm not sure if there is some subtlety I'm not thinking about.
```
In [4]: text = "a b\r\n1 2\r"

In [5]: ascii.read(text)
Out[5]: 
<Table length=1>
  a     b  
int64 int64
----- -----
    1     2
```

Reply:
Fixed